### DISCUSSION TITLE
`FabricDefaultAttributeRegistry#register` with an already built `DefaultAttributeContainer`

### DISCUSSION THREAD

**Discussion by triphora** [FabricMC/fabric-api#2587] [Mod Dev Support] [answered]
Hi,

Is there a possibility of a [`FabricDefaultAttributeRegistry#register`](https://github.com/FabricMC/fabric/blob/1.19.2/fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/entity/FabricDefaultAttributeRegistry.java#L51-L73) that takes an instance of [`DefaultAttributeContainer`](https://maven.fabricmc.net/docs/yarn-1.19.2+build.22/net/minecraft/entity/attribute/DefaultAttributeContainer.html) instead of [`DefaultAttributeContainer.Builder`](https://maven.fabricmc.net/docs/yarn-1.19.2+build.22/net/minecraft/entity/attribute/DefaultAttributeContainer.Builder.html)?

Currently the only registering API [builds it itself](https://github.com/FabricMC/fabric/blob/1.19.2/fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/entity/FabricDefaultAttributeRegistry.java#L70), but this is somewhat inconducive to cross-loader development as Forge's API takes in an already built `DefaultAttributeContainer`.

I personally cannot see a reason why not to add this - alas I am not experienced with entities, so there could be some other issue with doing this that I cannot see. I am willing to contribute this API myself, but I would rather make sure there aren't any reasons not to do so before spending the time to. :)

**Comment by apple502j** [comment, marked as answer]
Sounds great, could you PR it or make an issue? @wafflecoffee

**Reply by triphora** [reply]
Done! https://github.com/FabricMC/fabric/pull/2589
